hw6-atm/src/main/java/ru/otus/hw6/ATM.java

```diff
@@ -0,0 +1,24 @@
+package ru.otus.hw6;
+
+import java.util.Map;
+import java.util.Set;
+
+public interface ATM {
+    Map<BillValue, Integer> addCash(Map<BillValue, Integer> bills);
+
+    Map<BillValue, Integer> getCash(int sum);
+
+    Set<Cell> getCells();
+
+    int getBalanceSum();
+
+    void getBalance(Visitor visitor);
+
+    int getNumber();
+
+    Memento saveState();
+
+    ATM restoreState(Memento memento);

```

Comment:
Лучше так.
```suggestion
    void restoreState(Memento memento);
```
ATM должен восстановить свое состояние, а не вернуть какой-то другой ATM с исходным состоянием.
(впрочем это связано с замечанием ниже по DepartmentATM.restoreState())